packages/iov-socket/src/queueingstreamingsocket.ts

```diff
@@ -32,6 +35,13 @@ export class QueueingStreamingSocket {
     this.socket.disconnect();
   }
 
+  public reconnect(): void {
+    this.socket = new StreamingSocket(this.url, this.timeout);
+    this.events.imitate(this.socket.events);
```

Comment:
Nice find with `imitate`.

However, I guess we need some moderation here. For example when there is a connection failure, we get stream error events that should not be passed to the caller. So I suggest creating a stream event producer for `this.events` and use changing source streams to produce events.

Reply:
Should we just filter out all the errors? Are there any errors we do want to propagate? If not, we should be able to use the following (with a corresponding line in the constructor):
```ts
this.events.imitate(this.socket.events.replaceError(Stream.never));
```

Reply:
I'm not entirely sure how this will look at the end of the day. A custom producer os most flexible, but a bit more code. `this.events.imitate(this.socket.events.replaceError(Stream.never));` is probably sufficient for now

Reply:
What I am also concerned about is completion of the source stream

Reply:
Good point, we can do a similar move there though.